lib/tbot/tshwrap/wrap.go

```diff
@@ -130,22 +130,28 @@ type destinationHolder interface {
 	GetDestination() bot.Destination
 }
 
-// GetDestinationDirectory attempts to select an unambiguous destination, either from
-// CLI or YAML config. It returns an error if the selected destination is
-// invalid.
-func GetDestinationDirectory(botConfig *config.BotConfig) (*config.DestinationDirectory, error) {
+// GetDestinationDirectory attempts to select an unambiguous destination, either
+// from CLI or YAML config. It returns an error if the selected destination is
+// invalid. Note that CLI destinations will not be validated.
+func GetDestinationDirectory(cliDestinationPath string, botConfig *config.BotConfig) (*config.DestinationDirectory, error) {
+	if cliDestinationPath != "" {
+		d := &config.DestinationDirectory{
+			Path: cliDestinationPath,
+		}
+		if err := d.CheckAndSetDefaults(); err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		return d, nil
+	}
+
 	var destinationHolders []destinationHolder
 	for _, svc := range botConfig.Services {
 		if v, ok := svc.(destinationHolder); ok {
 			destinationHolders = append(destinationHolders, v)
 		}
 	}
-	// WARNING:
-	// This code is dependent on some unexpected "behavior" in
-	// config.FromCLIConf() - when users provide --destination-dir then all
-	// outputs configured in the YAML file are overwritten by an identity
-	// output with a directory destination with a path of --destination-dir.
-	// See: https://github.com/gravitational/teleport/issues/27206
```

Comment:
I removed this warning, I think #27206 was more or less resolved after this and #47130?